widget/embedded/src/utils/wallets.ts

```diff
@@ -415,98 +359,70 @@ export const getKeplrCompatibleConnectedWallets = (
   );
 };
 
-export function getTokensWithBalance(
-  tokens: TokenWithBalance[],
-  connectedWallets: ConnectedWallet[]
-): TokenWithBalance[] {
-  return tokens.map(({ balance, ...otherProps }) => {
-    const tokenAmount = numberToString(
-      new BigNumber(
-        getBalanceFromWallet(
-          connectedWallets,
-          otherProps.blockchain,
-          otherProps.symbol,
-          otherProps.address
-        )?.amount || ZERO
-      )
-    );
+export function getBalanceFor(
+  token: Token,
+  balances: TokensBalance
+): Balance | null {
+  const balance =
+    balances[`${token.symbol}-${token.blockchain}-${token.address ?? ''}`];
+  return balance ?? null;
+}
 
-    let tokenUsdValue = '';
-    if (otherProps.usdPrice) {
-      tokenUsdValue = numberToString(
-        new BigNumber(
-          getBalanceFromWallet(
-            connectedWallets,
-            otherProps.blockchain,
-            otherProps.symbol,
-            otherProps.address
-          )?.amount || ZERO
-        ).multipliedBy(otherProps.usdPrice)
-      );
-    }
+export function formatBalance(
+  token: Token,
+  balances: TokensBalance
+): Balance | null {
```

Comment:
I think maybe it's better to pass Balance to this method instead.

formatBalance(balance: Balance) -> Balance